### PR TITLE
Fix for accuracy calculation

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fix for accuracy (also confusion matrix had a problem) calculation in native as discussed in slack :)
Also added test against sklearn.
@borda, @williamFalcon, @justusschock 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/metrics/functional/test_classification.py [hunk 1]
```diff
@@ -23,6 +24,27 @@
...
+    (partial(fbeta_score_sk, average='macro', beta=2), partial(fbeta_score, beta=2)),
+    (confusion_matrix_sk, confusion_matrix)
+])
+def test_against_sklearn(sklearn_metric, torch_metric):
+    pred = torch.randint(10, (500,))
+    target = torch.randint(10, (500,))
+
+    assert torch.allclose(
+        torch.tensor(sklearn_metric(target, pred), dtype=torch.float),
+        torch_metric(pred, target))
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py:46]:
```suggestion
from sklearn.metrics import (
    accuracy_score as sk_accuracy,
    precision_score as sk_precision,
    recall_score as sk_recall,
    f1_score as sk_f1_score,
    fbeta_score as sk_fbeta_score,
    confusion_matrix as sk_confusion_matrix
)


@pytest.mark.parametrize(['sklearn_metric', 'torch_metric'], [
    (sk_accuracy, accuracy),
    (partial(sk_precision, average='macro'), precision),
    (partial(sk_recall, average='macro'), recall),
    (partial(sk_f1_score, average='macro'), f1_score),
    (partial(sk_fbeta_score, average='macro', beta=2), partial(fbeta_score, beta=2)),
    (sk_confusion_matrix, confusion_matrix)
])
def test_against_sklearn(sklearn_metric, torch_metric):
    """Compare PL metrics to sklearn version."""
    pred = torch.randint(10, (500,))
    target = torch.randint(10, (500,))

    assert torch.allclose(
        torch.tensor(sklearn_metric(target, pred), dtype=torch.float),
        torch_metric(pred, target))
```

File: tests/metrics/functional/test_classification.py [hunk 2]
```diff
@@ -113,40 +135,46 @@ def test_get_num_classes(pred, target, num_classes, expected_num_classes):
...
+    assert torch.allclose(expected_support.to(sup), sup)
 
 
 def test_multilabel_accuracy():
     # Dense label indicator matrix format
     y1 = torch.tensor([[0, 1, 1], [1, 0, 1]])
     y2 = torch.tensor([[0, 0, 1], [1, 0, 1]])
 
-    assert torch.allclose(accuracy(y1, y2, reduction='none'), torch.tensor([0.8333333134651184] * 2))
+    assert torch.allclose(accuracy(y1, y2, reduction='none'), torch.tensor([0.6666666666667, 1.]))
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py:177]:
```suggestion
    assert torch.allclose(accuracy(y1, y2, reduction='none'), torch.tensor([2 / 3, 1.]))
```

File: tests/metrics/functional/test_classification.py [hunk 3]
```diff
@@ -113,40 +135,46 @@ def test_get_num_classes(pred, target, num_classes, expected_num_classes):
...
+    assert sup.item() == expected_support
 
 
-@pytest.mark.parametrize(['pred', 'target', 'expected_tp', 'expected_fp', 'expected_tn', 'expected_fn'], [
+@pytest.mark.parametrize(['pred', 'target', 'expected_tp', 'expected_fp',
+                          'expected_tn', 'expected_fn', 'expected_support'], [
     pytest.param(torch.tensor([0., 2., 4., 4.]), torch.tensor([0., 4., 3., 4.]),
-                 [1, 0, 0, 0, 1], [0, 0, 1, 0, 1], [3, 4, 3, 3, 1], [0, 0, 0, 1, 1]),
+                 torch.tensor([1, 0, 0, 0, 1]), torch.tensor([0, 0, 1, 0, 1]), torch.tensor([3, 4, 3, 3, 1]),
+                 torch.tensor([0, 0, 0, 1, 1]), torch.tensor([1, 0, 0, 1, 2])),
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py:157]:
```suggestion
                 [1, 0, 0, 0, 1], [0, 0, 1, 0, 1], [3, 4, 3, 3, 1], [0, 0, 0, 1, 1], [1, 0, 0, 1, 2]),
```

File: tests/metrics/functional/test_classification.py [hunk 4]
```diff
@@ -113,40 +135,46 @@ def test_get_num_classes(pred, target, num_classes, expected_num_classes):
...
     pytest.param(to_onehot(torch.tensor([0., 2., 4., 4.])), torch.tensor([0., 4., 3., 4.]),
-                 [1, 0, 0, 0, 1], [0, 0, 1, 0, 1], [3, 4, 3, 3, 1], [0, 0, 0, 1, 1])
+                 torch.tensor([1, 0, 0, 0, 1]), torch.tensor([0, 0, 1, 0, 1]), torch.tensor([3, 4, 3, 3, 1]),
+                 torch.tensor([0, 0, 0, 1, 1]), torch.tensor([1, 0, 0, 1, 2]))
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py:160]:
```suggestion
                 [1, 0, 0, 0, 1], [0, 0, 1, 0, 1], [3, 4, 3, 3, 1], [0, 0, 0, 1, 1], [1, 0, 0, 1, 2])
```

File: tests/metrics/functional/test_classification.py [hunk 5]
```diff
@@ -113,40 +135,46 @@ def test_get_num_classes(pred, target, num_classes, expected_num_classes):
...
 
-    assert torch.allclose(torch.tensor(expected_tp).to(tp), tp)
-    assert torch.allclose(torch.tensor(expected_fp).to(fp), fp)
-    assert torch.allclose(torch.tensor(expected_tn).to(tn), tn)
-    assert torch.allclose(torch.tensor(expected_fn).to(fn), fn)
+    assert torch.allclose(expected_tp.to(tp), tp)
+    assert torch.allclose(expected_fp.to(fp), fp)
+    assert torch.allclose(expected_tn.to(tn), tn)
+    assert torch.allclose(expected_fn.to(fn), fn)
+    assert torch.allclose(expected_support.to(sup), sup)
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py:169]:
```suggestion
    assert torch.allclose(torch.tensor(expected_tp).to(tp), tp)
    assert torch.allclose(torch.tensor(expected_fp).to(fp), fp)
    assert torch.allclose(torch.tensor(expected_tn).to(tn), tn)
    assert torch.allclose(torch.tensor(expected_fn).to(fn), fn)
    assert torch.allclose(torch.tensor(expected_support).to(sup), sup)
```
